ipv-api/src/main/java/uk/gov/di/authentication/ipv/helpers/IPVCallbackHelper.java

```diff
@@ -0,0 +1,361 @@
+            AuthorisationCodeService authorisationCodeService,
+            CloudwatchMetricsService cloudwatchMetricsService,
+            ConfigurationService configurationService,
+            DynamoClientService dynamoClientService,
+            DynamoIdentityService dynamoIdentityService,
+            DynamoService dynamoService,
+            SerializationService objectMapper,
+            SessionService sessionService,
+            AwsSqsClient sqsClient) {
+        this.accountInterventionService = accountInterventionService;
+        this.auditService = auditService;
+        this.authCodeResponseService = authCodeResponseService;
+        this.authorisationCodeService = authorisationCodeService;
+        this.cloudwatchMetricsService = cloudwatchMetricsService;
+        this.configurationService = configurationService;
+        this.dynamoClientService = dynamoClientService;
+        this.dynamoIdentityService = dynamoIdentityService;
+        this.dynamoService = dynamoService;
+        this.objectMapper = objectMapper;
+        this.sessionService = sessionService;
+        this.sqsClient = sqsClient;
+    }
+
+    public APIGatewayProxyResponseEvent generateAuthenticationErrorResponse(
+            AuthenticationRequest authenticationRequest,
+            ErrorObject errorObject,
+            boolean noSessionErrorResponse,
+            String clientSessionId,
+            String sessionId) {
+        LOG.warn(
+                "Error in IPV AuthorisationResponse. ErrorCode: {}. ErrorDescription: {}. No Session Error: {}",
+                errorObject.getCode(),
+                errorObject.getDescription(),
+                noSessionErrorResponse);
+        auditService.submitAuditEvent(
+                IPVAuditableEvent.IPV_UNSUCCESSFUL_AUTHORISATION_RESPONSE_RECEIVED,
+                clientSessionId,
+                sessionId,
+                authenticationRequest.getClientID().getValue(),
+                AuditService.UNKNOWN,
+                AuditService.UNKNOWN,
+                AuditService.UNKNOWN,
+                AuditService.UNKNOWN,
+                AuditService.UNKNOWN);
+        var errorResponse =
+                new AuthenticationErrorResponse(
+                        authenticationRequest.getRedirectionURI(),
+                        errorObject,
+                        authenticationRequest.getState(),
+                        authenticationRequest.getResponseMode());
+        return generateApiGatewayProxyResponse(
+                302, "", Map.of(ResponseHeaders.LOCATION, errorResponse.toURI().toString()), null);
+    }
+
+    public AccountInterventionStatus getAccountInterventionStatus(
+            String internalPairwiseSubjectId) {
+        var accountInterventionStatus =
+                segmentedFunctionCall(
+                        "AIS: getAccountStatus",
+                        () ->
+                                this.accountInterventionService.getAccountStatus(
```

Comment:
This has been changed to also take in auditContext (optional) - something to include?

Reply:
What do you think? 🙂